django/db/models/fields/__init__.py

```diff
@@ -817,6 +817,28 @@ def get_db_converters(self, connection):
             return [self.from_db_value]
         return []
 
+    @cached_property
+    def non_db_attrs(self):
```

Comment:
This could just be a tuple, no need for a property here. Subclasses may opt to use a property or `cached_property`, but that's no concern in the base class.